### PR TITLE
update vehicle status intent message generation format 

### PR DESCRIPTION
**Pull request by dan-du-car**
# PR Details
## Description
update vehicle status intent message generation format 

## Related Issue
NA

## How Has This Been Tested?
Local integration testing

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: message_services/src/models/vehicle_status_intent.cpp [hunk 1]
```diff
@@ -85,65 +85,68 @@ namespace message_services
...
+                writer->String("timestamp");
+                writer->Uint64(this->getCur_timestamp());
+                writer->EndObject();
+                writer->String("payload");
                 writer->StartObject();
                 writer->String("v_id");
                 writer->String(this->getVehicle_id().c_str());
                 writer->String("v_length");
-                writer->String(std::to_string(this->getVehicle_length()).c_str());
+                writer->Uint64(this->getVehicle_length());
```
Comment (Saeid-Solei) [inline message_services/src/models/vehicle_status_intent.cpp:99]: I think the unit of the vehicle length should be double if it is in meter. The vehicle length could be 4.5 meters for example.

File: message_services/src/models/vehicle_status_intent.cpp [hunk 2]
```diff
@@ -85,65 +85,68 @@ namespace message_services
...
-                writer->String("cur_timestamp");
-                writer->String(std::to_string(this->getCur_timestamp()).c_str());
-                writer->String("enter_lanelet_id");
-                writer->String(std::to_string(this->getEnter_lanelet_id()).c_str());
+                writer->Double(this->getCur_distance());
+                writer->String("entry_lane_id");
+                writer->Uint64(this->getEnter_lanelet_id());
                 writer->String("dest_lane_id");
-                writer->String(std::to_string(this->getDest_lanelet_id()).c_str());
+                writer->Uint64(this->getDest_lanelet_id());
```
Comment (Saeid-Solei) [inline message_services/src/models/vehicle_status_intent.cpp:123]: In the estimated path, the type of lane id is defined as Int64. Just wondering if it is different than Uint64 or not.

File: message_services/src/models/vehicle_status_intent.cpp [hunk 3]
```diff
@@ -85,65 +85,68 @@ namespace message_services
...
                 writer->String("depart_pos");
-                writer->String(std::to_string(this->getDepart_position()).c_str());
-                writer->String("is_allowed_int");
-                writer->String(std::to_string(this->getIs_allowed()).c_str());
+                writer->Uint64(this->getDepart_position());
+                writer->String("is_allowed");
+                writer->Bool(this->getIs_allowed());
                 writer->String("cur_lanelet_id");
-                writer->String(std::to_string(this->getCur_lanelet_id()).c_str());
+                writer->Uint64(this->getCur_lanelet_id());
```
Comment (Saeid-Solei) [inline message_services/src/models/vehicle_status_intent.cpp:117]: change it to Int64.

File: message_services/src/models/vehicle_status_intent.cpp [hunk 4]
```diff
@@ -85,65 +85,68 @@ namespace message_services
...
                 writer->String("cur_ds");
-                writer->String(std::to_string(this->getCur_distance()).c_str());
-                writer->String("cur_timestamp");
-                writer->String(std::to_string(this->getCur_timestamp()).c_str());
-                writer->String("enter_lanelet_id");
-                writer->String(std::to_string(this->getEnter_lanelet_id()).c_str());
+                writer->Double(this->getCur_distance());
+                writer->String("entry_lane_id");
+                writer->Uint64(this->getEnter_lanelet_id());
```
Comment (Saeid-Solei) [inline message_services/src/models/vehicle_status_intent.cpp:121]: change it to Int64.

File: message_services/src/models/vehicle_status_intent.cpp [hunk 5]
```diff
@@ -85,65 +85,68 @@ namespace message_services
...
-                writer->String(std::to_string(this->getLink_lanelet_id()).c_str());
-                writer->String("earliest_stop_t");
-                writer->String(std::to_string(this->getEarliest_stop_timestamp()).c_str());
-                writer->String("actual_stop_t");
-                writer->String(std::to_string(this->getActual_stop_timestamp()).c_str());
-                writer->String("actual_enter_t");
-                writer->String(std::to_string(this->getActual_enter_timestamp()).c_str());
-                writer->String("actual_depart_t");
-                writer->String(std::to_string(this->getActual_depart_timestamp()).c_str());
+                writer->Uint64(this->getLink_lanelet_id());
```
Comment (Saeid-Solei) [inline message_services/src/models/vehicle_status_intent.cpp:125]: Change it to Int64.

File: message_services/src/models/vehicle_status_intent.cpp (hunk 1 above)
Comment (dan-du-car) [reply message_services/src/models/vehicle_status_intent.cpp:99]: https://github.com/usdot-fhwa-stol/carma-messenger/blob/564833996efad24056cc1836a101640a84791d3b/carma-messenger-core/cpp_message/include/generated/VehicleWidth.h#L22

Comment (dan-du-car) [reply message_services/src/models/vehicle_status_intent.cpp:99]: https://github.com/usdot-fhwa-stol/carma-messenger/blob/564833996efad24056cc1836a101640a84791d3b/carma-messenger-core/cpp_message/include/generated/VehicleLength.h#L22

**Review by Saeid-Solei — CHANGES_REQUESTED** [review]

File: message_services/src/models/vehicle_status_intent.cpp (hunk 4 above)
Comment (dan-du-car) [reply message_services/src/models/vehicle_status_intent.cpp]: https://github.com/usdot-fhwa-stol/carma-streets/pull/31/commits/ecff51587f2d774676eebd4c355599b660b33eb7

File: message_services/src/models/vehicle_status_intent.cpp (hunk 5 above)
Comment (dan-du-car) [reply message_services/src/models/vehicle_status_intent.cpp]: https://github.com/usdot-fhwa-stol/carma-streets/pull/31/commits/ecff51587f2d774676eebd4c355599b660b33eb7

File: message_services/src/models/vehicle_status_intent.cpp (hunk 2 above)
Comment (dan-du-car) [reply message_services/src/models/vehicle_status_intent.cpp]: https://github.com/usdot-fhwa-stol/carma-streets/pull/31/commits/ecff51587f2d774676eebd4c355599b660b33eb7

**Review by Saeid-Solei — APPROVED** [review]

**Pull request merged**